tools/dependency/dump_repository_locations.py

```diff
@@ -0,0 +1,31 @@
+import json
+import pathlib
+import sys
+
+
+def _format_version(s, version):
+    return s.format(
+        version=version,
+        dash_version=version.replace(".", "-"),
+        underscore_version=version.replace(".", "_"))
+
+
+def main(*repo_location_files):
+    locations = {}
+
+    for repo_file in repo_location_files:
+        repo_locations = json.loads(pathlib.Path(repo_file).read_text())
+        for location in repo_locations.values():
+            if "version" in location:
+                if "strip_prefix" in location:
+                    location["strip_prefix"] = _format_version(
+                        location["strip_prefix"], location["version"])
+                location["urls"] = [
+                    _format_version(url, location["version"]) for url in location["urls"]
+                ]
```

Comment:
Why can't this be done more intelligently during the generation of the input JSON files? 

I used to have some Python code that was usable in both Bazel and Python in the `.bzl` and that centralized this logic.

Reply:
so, bazel can provide `json` files, but we need a couple of extra steps here - interpolating the urls/versions, and aggregating from several `repository_locations.bzl` files

as mentioned in the ticket comment i have a more generic solution for this based on `data` providers, and a mutation pipeline - it was designed to work in several places, here being one of them

the old code that used `SourceFileLoader` was problematic for a few reasons - eg you cant put `load` files in the `.bzl` - ultimately i dont think its a good idea to parse starlark using a python parser

also, it wasnt being done in one place, i have been gradually unpicking the places that used this pattern, firstly the docs, and now here - removal is almost complete

Reply:
the way the `SourceFileLoader` code is setup it causes another problem also, the paths are different when a job is `run` or `build` 

shifting this to a `py_binary` with `data` and `location` `args` kinda resolves that

Reply:
ok, so i think i understand better your suggestion

i have made the `json_data` write rules for `repository_locations` do the interpolation in bazel using the existing bazel util

this removes the need to do the mangling in python, but also the need to load any starlark as python

Reply:
to make it work i also had to add a `merge_dicts` util as there is no one-liner with starlark python to merge dicts in place

Reply:
wh00t, thanks!